### ISSUE TITLE
Duplicate providers with multiple app bootstrap

### ISSUE THREAD

**Issue by Exlord** [angular/angular.js#11961]
I have a project that has 2 ng-app running, the second one is bootstrapped manually.
I was having a problem with routing (http://stackoverflow.com/questions/30479809/angular-multi-spa-with-routing) and while going throw the source code to find the problem, i noticed that the ```$browser``` provider is creating a whole new instance of the ```browser``` object when the second app is bootstrapped .

It was my understanding that the providers are singleton objects so it should not make a difference when bootstrapping a new app. should id?

Since ```$browser``` is watching the browsers url changes, they were falling into a infinite  digest loop.

I made a gist for the codes.
I am using angular 1.3.15, and it also needs jquery and angular-route.
https://gist.github.com/Exlord/1fb2a8949e2d1850d234

page1.html contains a new angular module. index.html loads the page1 on link click and bootstraps its app

**Comment by pkozlowski-opensource** [comment]
All objects in the DI container are singletons, but each application gets a new injector (DI container). So you will have a separate $browser objects living in memory for each AngularJS application.

What you can do in your case is to try to override $browser service with a custom (noop) object for one of application. But bootstraping multiple applications on one page is problematic as there is some global state (location, cookies etc.) where 2 application will "clash".

In short: bootstraping several AngularJS applications on one HTML page is currently problematic and not advisable.

**Comment by Exlord** [comment]
I fix my problem with this : 
https://gist.github.com/Exlord/1fc5cd3f3c8d2eccdb5d
